components/admin_console/filter/team_filter_dropdown/team_filter_dropdown.tsx

```diff
@@ -0,0 +1,328 @@
+    (selectedTeamIds: string[], teams: Team[]) => teams,
+    (selectedTeamIds, teams) => teams.filter((team) => selectedTeamIds.includes(team.id)),
+);
+
+const TEAMS_PER_PAGE = 5;
+const MAX_BUTTON_TEXT_LENGTH = 25;
+class TeamFilterDropdown extends React.PureComponent<Props, State> {
+    private ref: React.RefObject<HTMLDivElement>;
+
+    public constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            page: 0,
+            loading: false,
+            show: false,
+            savedSelectedTeams: [],
+            searchResults: [],
+            searchTerm: '',
+            searchTotal: 0,
+        };
+
+        this.ref = React.createRef();
+    }
+
+    componentDidMount() {
+        document.addEventListener('mousedown', this.handleClickOutside);
+        this.props.actions.getData(0, TEAMS_PER_PAGE);
+    }
+
+    componentWillUnmount = () => {
+        document.removeEventListener('mousedown', this.handleClickOutside);
+    }
+
+    hidePopover = () => {
+        this.setState({show: false});
+    }
+
+    togglePopover = () => {
+        if (this.state.show) {
+            this.hidePopover();
+            return;
+        }
+
+        const selectedTeamIds = this.props.option.values.team_ids.value as string[];
+        const selectedTeams = getSelectedTeams(selectedTeamIds, this.props.teams);
+        const savedSelectedTeams = selectedTeams.sort((a, b) => a.display_name.localeCompare(b.display_name));
+        this.setState({show: true, savedSelectedTeams, searchTerm: ''});
+    }
+
+    handleClickOutside = (event: MouseEvent) => {
+        if (this.ref?.current?.contains(event.target as Node)) {
+            return;
+        }
+        this.hidePopover();
+    }
+
+    hasMore = (): boolean => {
+        if (this.state.searchTerm.length > 0) {
+            return !this.state.loading && this.state.searchTotal > this.state.searchResults.length;
+        }
+        return !this.state.loading && this.props.total > this.props.teams.length;
+    }
+
+    loadMore = async () => {
+        const {searchTerm} = this.state;
+        this.setState({loading: true});
+        const page = this.state.page + 1;
+        if (searchTerm.length > 0) {
+            this.searchTeams(searchTerm, page);
+        } else {
+            await this.props.actions.getData(page, TEAMS_PER_PAGE);
+        }
+        this.setState({page, loading: false});
+    }
+
+    searchTeams = async (term: string, page: number) => {
+        let searchResults = [];
+        let searchTotal = 0;
+        const response = await this.props.actions.searchTeams(term, page, TEAMS_PER_PAGE);
+        if (response?.data) {
+            searchResults = page > 0 ? this.state.searchResults.concat(response.data.teams) : response.data.teams;
+            searchTotal = response.data.total_count;
+        }
+        this.setState({page, loading: false, searchResults, searchTotal, searchTerm: term});
+    }
```

Comment:
What about error handling for `loadMore` and `searchTeams`? What happens in the UI?

Reply:
If this fails it will just show `No results found`, which is sufficient for this case i think

Reply:
Recently in another PR this came up and UX was planning to do something consistent for all potential server errors I believe. Not sure if we're there yet. cc @hahmadia @michaelgamble do either of you know where that landed?

Reply:
In my case, if the search fails, instead of showing no results, Michael had me update the ticket to something like "The search failed. Please try again" instead of no results found. That's what we did and we made a a ticket to enhance the text by making the try again clickable and it will redo the search query. But that's as far as we got, I'm not sure if Michael has other thoughts other than what we discussed at that time.

Reply:
I'll discuss this with him in my ux review now